hack/trivy_scan.py

```diff
@@ -0,0 +1,318 @@
+        uniq_wg_images = sorted(wg_images)
+        all_images.update(uniq_wg_images)
+        save_images(wg, uniq_wg_images, version)
+
+    # Ensure uniqueness across all workgroups
+    uniq_images = sorted(all_images)
+    save_images("all", uniq_images, version)
+
+parser = argparse.ArgumentParser(description="Extract images from Kubeflow kustomizations.")
+# Define a positional argument 'version' with optional occurrence and default value 'latest'. You can run this file as python3 <filename>.py or python <filename>.py <version>
+parser.add_argument("version", nargs="?", type=str, default="latest", help="Kubeflow version to use (defaults to latest).")
+args = parser.parse_args()
+extract_images(args.version)
+
+
+
+print("Started scanning images")
+
+# Get list of text files excluding "kf_latest_all_images.txt"
+files = [f for f in glob.glob(os.path.join(DIRECTORY, "*.txt")) if not f.endswith("kf_latest_all_images.txt")]
+
+# Loop through each text file in the specified directory
+for file in files:
+    print(f"Scanning images in {file}")
+
+    file_base_name = os.path.basename(file).replace('.txt', '')
+
+    # Directory to save reports for this specific file
+    file_reports_dir = os.path.join(SCAN_REPORTS_DIR, file_base_name)
+    os.makedirs(file_reports_dir, exist_ok=True)
+
+    # Directory to save security count
+    severity_count = os.path.join(file_reports_dir, "severity_counts")
+    os.makedirs(severity_count, exist_ok=True)
+
+    with open(file, 'r') as f:
+        lines = f.readlines()
+
+    for line in lines:
+        line = line.strip()
+        image_name = line.split(':')[0]
+        image_tag = line.split(':')[1] if ':' in line else ''
+
+        image_name_scan = image_name.split('/')[-1]
+
+        if image_tag:
+            image_name_scan = f"{image_name_scan}:{image_tag}"
+
+        print(f"Scanning {image_name_scan}")
+
+        scan_output_file = os.path.join(file_reports_dir, f"{image_name_scan}_scan.json")
+
+        try:
+            subprocess.run(["trivy", "image", "--format", "json", "--output", scan_output_file, line], check=True)
+
+            with open(scan_output_file, 'r') as json_file:
+                scan_data = json.load(json_file)
+
+            if not scan_data.get('Results'):
+                print(f"No vulnerabilities found in {image_name}:{image_tag}")
+            else:
+                vulnerabilities = [
+                    result['Vulnerabilities']
+                    for result in scan_data['Results']
+                    if 'Vulnerabilities' in result and result['Vulnerabilities']
+                ]
+
+                if not vulnerabilities:
+                    print(f"The vulnerability detection may be insufficient because security updates are not provided for {image_name}:{image_tag}")
+                else:
+                    severity_counts = {"LOW": 0, "MEDIUM": 0, "HIGH": 0, "CRITICAL": 0}
+                    for vulns in vulnerabilities:
```

Comment:
Please consider proper names. e.g. for vulnerability in vulnerabilities for the follow up PR.